firefox-ios/Client/Frontend/Browser/BrowserViewController/Extensions/BrowserViewController+WebViewDelegates.swift

```diff
@@ -646,6 +646,15 @@ extension BrowserViewController: WKNavigationDelegate {
                 }
             }
 
+            // Note: When "Undo" a tab with a Youtube video, navigationAction.request
+            // return the Youtube Homepage URL instead the video URL,
+            // even if the tab have the correct URL.
+            if let tabURL = tab.url, navigationAction.isIncorrectYoutubeURL(tabURL: tabURL) {
+                webView.load(URLRequest(url: tabURL))
+                decisionHandler(.cancel)
+                return
+            }
```

Comment:
hmm this doesn't seem right, if a tab is being closed and user taps on undo then we usually have two options 
a) fully close the tab
b) partial close the tab

In both cases we should have the url to load and not worry about redirection request from youtube.
Also if the request is coming from a website itself then we shouldn't be the one handling it manually here.

Could you look into the actual undo logic and why the url that was previously there doesn't load?

Reply:
DM me if you need any pointers